### PR TITLE
Investigate inconsistencies in screenshots

### PR DESCRIPTION
**Pull request by SkullTech**
Added a script to extract inconsistencies present in the screenshot archive, and produce a `csv` named `inconsistencies.csv`. The columns of the CSV being  
- WEBCOMPAT-ID
- ELEMENT-ID (Can be blank, representing PNGs of format `XXXX_browser.png`)
- FIREFOX (Presence of Firefox screenshot)
- CHROME (Presence of Chrome screenshot)

### CODE REVIEW

**Comment by SkullTech** [comment]
Resolves #10

**Comment by SkullTech** [comment]
Running the script

```console
> python3 data_inconsistencies.py
[*] Getting inconsistencies in screenshots...
[*] 5758 inconsistencies found.
[*] Writing to inconsistencies.csv... Done!
```

Sample lines from the generated `inconsistencies.csv`
```
WEBCOMPAT-ID,ELEMENT-ID,FIREFOX,CHROME
8855,m_login_password_1,True,False
8855,did_submit_4,True,False
8855,login_identify_search_placeholder_5,True,False
8855,did_submit_3,True,False
8855,forgot-password-link_2,True,False
```

**Comment by marco-c** [comment]
I forgot to mention that the actual format of the name also contains a sequence number.
So it is either WEBCOMPAT-ID_ELEMENT-ID_SEQUENCE-NUM_BROWSER.png or WEBCOMPAT-ID_BROWSER.png.

We should sort the CSV results by WEBCOMPAT-ID first and then by SEQUENCE-NUM.

**Comment by SkullTech** [comment]
Done!

File: data_inconsistencies.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import os
+import csv
+
+
+def get_inconsistencies():
+    files = os.listdir('./data/')
+
+    parsed = {}
+    for file in files:
```
Comment (marco-c) [inline data_inconsistencies.py]: Can you rename the `file` variable? It's a keyword in Python, so it gets highlighted differently than other variables (which can be confusing).

File: data_inconsistencies.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def main():
+    print('[*] Getting inconsistencies in screenshots...')
+    incons = get_inconsistencies()
+    print('[*] {} inconsistencies found.'.format(len(incons)))
+
+    print('[*] Writing to inconsistencies.csv... ', end='')
+    with open('inconsistencies.csv', 'w') as csvfile:
+            writer = csv.writer(csvfile, delimiter=',')
```
Comment (marco-c) [inline data_inconsistencies.py]: Too many spaces here

File: data_inconsistencies.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def get_inconsistencies():
+    files = os.listdir('./data/')
+
+    parsed = {}
+    for file in files:
+        parts = os.path.splitext(file)[0].split('_')
+
+        ID = int(parts[0])
```
Comment (marco-c) [inline data_inconsistencies.py]: Same with ID, let's use webcompatID.

File: data_inconsistencies.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        if ID not in parsed:
+            parsed[ID] = {}
+        element = '_'.join(parts[1:-1])
+        if element:
```
Comment (marco-c) [inline data_inconsistencies.py]: You could avoid this variable here and use the length of parts. If there are two parts, it's the short format, if there are more it's the other format.

File: data_inconsistencies.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            spl = element.split('_')
```
Comment (marco-c) [inline data_inconsistencies.py]: You can probably reuse `parts` here.

File: data_inconsistencies.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        parsed[ID][(element, sequence)].append(parts[-1])
+
+    incons = []
+    for key, value in parsed.items():
+        for (element, sequence), browsers in value.items():
+            if len(browsers) < 2:
+                incons.append([key, element, sequence, 'firefox' in browsers, 'chrome' in browsers])
+
+    incons.sort(key=lambda x: x[2])
+    incons.sort(key=lambda x: x[0])
```
Comment (marco-c) [inline data_inconsistencies.py]: Nit: you can probably sort in one line if the key lambda returns a tuple instead

File: data_inconsistencies.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    return incons
+
+
+def main():
+    print('[*] Getting inconsistencies in screenshots...')
+    incons = get_inconsistencies()
+    print('[*] {} inconsistencies found.'.format(len(incons)))
+    print('[*] Writing to inconsistencies.csv... ', end='')
+    with open('inconsistencies.csv', 'w') as csvfile:
+            writer = csv.writer(csvfile, delimiter=',')
```
Comment (marco-c) [inline data_inconsistencies.py]: Too much indentation

**Pull request merged**